### PR TITLE
Mwpw 136767 prevent scroll

### PR DESCRIPTION
**Pull request by saragajic11**
This PR should resolve the issue of scrollable background when modal dialog (which is authored to have a 'curtain') is opened. In other cases (with 'curtain-off') background should be scrollable per requirements.

Resolves: [MWPW-136767](https://jira.corp.adobe.com/browse/MWPW-136767)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/rclayton/video-examples?martech=off
- After: https://mwpw-136767-prevent-scroll--milo--saragajic11.hlx.page/drafts/rclayton/video-examples?martech=off

Commerce dialog (it should work the same as before this PR):
- Before: https://main--milo--adobecom.hlx.page/drafts/ruchika/discover1?martech=off
- After: https://mwpw-136767-prevent-scroll--milo--saragajic11.hlx.page/drafts/ruchika/discover1?martech=off

### CODE REVIEW

File: libs/blocks/modal/modal.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ function closeModal(modal) {
 
   document.querySelectorAll(`#${id}`).forEach((mod) => {
     if (mod.nextElementSibling?.classList.contains('modal-curtain')) {
+      document.querySelector('body').style.removeProperty('overflow');
```
Comment (Blainegunn) [inline libs/blocks/modal/modal.js:46]: I would do it like this:
```
document.body.style.removeProperty('overflow');
// and 
document.body.style.overflow = 'hidden';
slightly faster in my observation.

Comment (saragajic11) [reply libs/blocks/modal/modal.js]: Yes, makes sense, resolved here: https://github.com/adobecom/milo/pull/1419/commits/07dd951b1c25735d6a519b3e013f699a854e3a3d
thanks @Blainegunn !

**Review by JasonHowellSlavin — APPROVED** [review]

**Review by Ruchika4 — APPROVED** [review]

**Review by fullcolorcoder — COMMENTED** [review]
Would it be more true to purpose to add a open class to the body and then do this with CSS?

**Review by chrischrischris — CHANGES_REQUESTED** [review]
Can you coordinate with @Ruchika4 to use `disable-scroll` and take `commerce-modal-open` out?  There is no need to have two classes that do the exact same thing.

See
https://github.com/adobecom/milo/pull/1234
https://github.com/adobecom/milo/pull/1285

**Comment by Ruchika4** [comment]
> Can you coordinate with @Ruchika4 to use `disable-scroll` and take `commerce-modal-open` out? There is no need to have two classes that do the exact same thing.
> 
> See #1234 #1285

Thanks for catching that @chrischrischris . I'll take a look

**Comment by Ruchika4** [comment]
> > Can you coordinate with @Ruchika4 to use `disable-scroll` and take `commerce-modal-open` out? There is no need to have two classes that do the exact same thing.
> > See #1234 #1285
> 
> Thanks for catching that @chrischrischris . I'll take a look

Had a discussion with @saragajic11 for the required changes. She'll make the changes and update. cc @chrischrischris

**Review by fullcolorcoder — APPROVED** [review]

**Comment by saragajic11** [comment]
Hi @Ruchika4 @chrischrischris , I have updated this PR as agreed: https://github.com/adobecom/milo/pull/1419/commits/ae01e292851e97dcdd4437ce8b9b497beb295ed1

**Comment by saragajic11** [comment]
Hi @chrischrischris @Blainegunn , after merging main this morning to my branch, footer got shifted on my branch's pages
(example page: https://mwpw-136767-prevent-scroll--milo--saragajic11.hlx.page/drafts/rclayton/video-examples?martech=off) . Footer is shifted because .feds-footer-icons div is displayed (even though I don't have any footer related changes in my PR). I created a new branch on my fork from fresh main just to confirm it has nothing to do with my changes and it looks good:
https://mwpw-136767-disable-scroll--milo--saragajic11.hlx.page/drafts/rclayton/video-examples?martech=off 
Should I create a new PR from that branch?

**Comment by Ruchika4** [comment]
> * https://mwpw-136767-prevent-scroll--milo--saragajic11.hlx.page/drafts/rclayton/video-examples?martech=off

Thanks @saragajic11 . @spadmasa could you please help test it for CC pages and make sure scroll is working as expected when commerce modal is open. Thanks

**Comment by spadmasa** [comment]
> > * https://mwpw-136767-prevent-scroll--milo--saragajic11.hlx.page/drafts/rclayton/video-examples?martech=off
> 
> Thanks @saragajic11 . @spadmasa could you please help test it for CC pages and make sure scroll is working as expected when commerce modal is open. Thanks

@Ruchika4  the flow has been verified in cc pages for commerce modal 
https://main--cc--adobecom.hlx.page/creativecloud/animation/discover?milolibs=mwpw-136767-prevent-scroll--milo--saragajic11
popup is loaded fine and background is not seen as scrollable
image

but noticed an issue of footer is being shifted, which is already intimated to Sara to take a look.
image

**Comment by Ruchika4** [comment]
@saragajic11 can we please close this PR as a new PR for the same thing has been raised here https://github.com/adobecom/milo/pull/1457

cc @Blainegunn

**Pull request closed without merging**